glean-core/src/metrics/mod.rs

```diff
@@ -165,7 +169,43 @@ pub trait MetricType {
     /// This depends on the metrics own state, as determined by its metadata,
     /// and whether upload is enabled on the Glean object.
     fn should_record(&self, glean: &Glean) -> bool {
-        glean.is_upload_enabled() && self.meta().should_record()
+        if !glean.is_upload_enabled() {
+            return false;
+        }
+
+        // Get the current disabled field from the metric metadata, including
+        // the encoded Nimbus epoch
+        let disabled_field = self.meta().disabled.load(Ordering::Relaxed);
```

Comment:
Something I wasn't entirely sure about, do I need `Ordering::Acquire` here instead of `::Relaxed`?

Reply:
I never really know. I'm usually conservative and go with `Acquire`.
But let's think here for a moment:

Most metrics recording happens on the dispatcher. That runs one task at a time. Thus `Relaxed` would be fine enough. There's no concurrent reading going on.

_Some_ metrics are record synchronously.
* This happens during startup -> No dispatcher tasks running yet.
* This happens during ping submission -> Itself runs synchronously during startup OR _also_ runs on the dispatcher, thus no other task concurrently.

Therefore I think `Relaxed` is good enough.